lesson 6/Tensor/app/src/main/cpp/native-lib.cpp

```diff
@@ -0,0 +1,95 @@
+#include <jni.h>
+#include <string>
+#include <string.h>
+#include <sys/socket.h>	//socket
+#include <arpa/inet.h>	//inet_addr
+#include <unistd.h>
+#include <android/log.h>
+#include <errno.h>
+#include <locale>
+
+
+#define LOG_TAG "CLIENT_APP"
+
+jbyteArray getImage(jobject context, JNIEnv* env) {
+    jclass contextThemeWrapperClass = env->FindClass("android/view/ContextThemeWrapper");
+    jclass assetManagerClass = env->FindClass("android/content/res/AssetManager");
+    jclass inputStreamClass = env->FindClass("java/io/InputStream");
+
+    jmethodID getAssetsID = env->GetMethodID(contextThemeWrapperClass, "getAssets", "()Landroid/content/res/AssetManager;");
+    jmethodID openID = env->GetMethodID(assetManagerClass, "open", "(Ljava/lang/String;)Ljava/io/InputStream;");
+    jmethodID availableID = env->GetMethodID(inputStreamClass, "available", "()I");
```

Comment:
Your implementation of this line was like this:
`jmethodID available = env->GetMethodID(InputStream, "available()", ()I);`
There are a few issues here:
1. Again you should name it `availableID`  not just `available`.
2. Second argument is the method name, so there shouldn't be any parenthesis, so `available`, not `available()`
3. You forgot to make the 3rd argument string with quotes `" "`